source/rainerscript/constant_strings.rst

```diff
@@ -1,15 +1,62 @@
-Constant Strings
+String Constants
 ================
 
+String constants are necessary in any scripting language. The provide
+values that never change during rsyslog's run.
+
+Uses
+----
 String constants are necessary in many places: comparisons,
 configuration parameter values and function arguments, to name a few
 important ones.
 
-In constant strings, special characters are escape by prepending a
+In string constants, special characters are escape by prepending a
 backslash in front of them -- just in the same way this is done in the C
 programming language or PHP.
 
 If in doubt how to properly escape, use the `RainerScript String Escape
 Online
 Tool <http://www.rsyslog.com/rainerscript-constant-string-escaper/>`_.
 
+Types
+-----
+
+Rsyslog provides different types of string constant, closely inspired
+by the shell:
+
+- single quotes
+
+  Values are used unaltered, execept for escape sequences, which are
+  escaped.
+
+- double quotes
+
+  Right now, equivalent to single quotes, but $ signs need to be escaped.
```

Comment:
Yup, next line 😛😉

Reply:
Evidently I'm blind, I see it now:

> If not escaped, a syntax error will be generated and rsyslog startup be aborted in most cases.

 Thanks!